Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -768,39 +763,21 @@ namespace Babylon
         delete info[0].As<Napi::External<VertexArray>>().Data();
     }
 
-    void NativeEngine::BindBuffer(const Napi::CallbackInfo& info)
-    {
-        VertexBufferData* vertexBufferData = info[0].As<Napi::External<VertexBufferData>>().Data();
-
-        const uint32_t location = info[1].As<Napi::Number>().Uint32Value();
-        const uint32_t byteStride = info[3].As<Napi::Number>().Uint32Value();
-        const uint32_t numElements = info[4].As<Napi::Number>().Uint32Value();
-        const uint32_t type = info[5].As<Napi::Number>().Uint32Value();
-        const bool normalized = info[6].As<Napi::Boolean>().Value();
-
-        bgfx::VertexLayout vertexLayout{};
-        vertexLayout.begin();
-        const bgfx::Attrib::Enum attrib = static_cast<bgfx::Attrib::Enum>(location);
-        const bgfx::AttribType::Enum attribType = ConvertAttribType(static_cast<WebGLAttribType>(type));
-        vertexLayout.add(attrib, static_cast<uint8_t>(numElements), attribType, normalized);
-        vertexLayout.m_stride = static_cast<uint16_t>(byteStride);
-        vertexLayout.end();
-
-        vertexBufferData->EnsureFinalized(info.Env(), vertexLayout);
-        vertexBufferData->SetAsBgfxVertexBuffer(static_cast<uint8_t>(location), 0);
-    }
-
     void NativeEngine::BindVertexArray(const Napi::CallbackInfo& info)
     {
         const auto& vertexArray = *(info[0].As<Napi::External<VertexArray>>().Data());
 
-        vertexArray.indexBuffer.data->SetBgfxIndexBuffer();
+        // a vertex array might not have an index buffer associated with
+        if (vertexArray.indexBuffer.data)
+        {
+            vertexArray.indexBuffer.data->SetBgfxIndexBuffer();
+        }
```

Comment:
Technically speaking, a VAO (at least for WebGL) doesn't include the index buffer. Maybe we should separate these into separate functions like it is in WebGL.

Reply:
I'm opening an issue for that. I don't like to destroy and recreate a new vertex array object when call bindbuffers: Too many allocations/deallocations.

Reply:
https://github.com/BabylonJS/BabylonNative/issues/361